setup.py

```diff
@@ -4,100 +4,106 @@
+    #
+    #  where {commitcount} is the number of commits after the tag (obtained with `git describe`)
+    #  and {tag+1} is the tag version incremented by one (i.e., if tag=0.0.1, it's 0.0.2). This
+    #  is needed because .dev tags compare as less than non-.dev tags, per PEP-440.
+    #
+    # * If there are no annotated tags in the past, the version is:
+    #
+    #    '0.0.0.dev{commitcount}+{gitsha}'
+    #
+    # Inspired by https://github.com/pyfidelity/setuptools-git-version
+    # Creates PEP-440 compliant versions
 
-      from subprocess import check_output
+    from subprocess import check_output
 
-      command = 'git describe --tags --long --dirty --always'
-      version = check_output(command.split()).decode('utf-8').strip()
+    command = "git describe --tags --long --dirty --always"
+    version = check_output(command.split()).decode("utf-8").strip()
 
-      parts = version.split('-')
-      if len(parts) in (3, 4):
-            dirty = len(parts) == 4
-            tag, count, sha = parts[:3]
-            if not tag.startswith('v'):
-                  raise Exception("Annotated tags on the repository must begin with the letter 'v'. Please fix this then try building agains.")
-            tag = tag[1:]
-            if count == '0' and not dirty:
-                  return tag
-      elif len(parts) in (1, 2):
-            tag = "0.0.0"
-            dirty = len(parts) == 2
-            sha = parts[0]
-            # Number of commits since the beginning of the current branch
-            count = check_output("git rev-list --count HEAD".split()).decode('utf-8').strip()
+    parts = version.split("-")
+    if len(parts) in (3, 4):
+        dirty = len(parts) == 4
+        tag, count, sha = parts[:3]
+        if not tag.startswith("v"):
+            raise Exception(
+                "Annotated tags on the repository must begin with the letter 'v'. Please fix this then try building agains."
+            )
+        tag = tag[1:]
+        if count == "0" and not dirty:
+            return tag
+    elif len(parts) in (1, 2):
+        tag = "0.0.0"
+        dirty = len(parts) == 2
+        sha = parts[0]
+        # Number of commits since the beginning of the current branch
+        count = (
+            check_output("git rev-list --count HEAD".split()).decode("utf-8").strip()
+        )
 
-      # this will be a .dev version; assume the version is of the form
-      # major[.minor[.patchlevel]], and increment the patchlevel by 1.
-      (major, minor, patchlevel) = (tag + ".0.0").split('.')[:3] # because minor and/or patchlevel are optional
-      patchlevel = str( int(patchlevel) + 1 )
-      tag = '.'.join((major, minor, patchlevel))
+    # this will be a .dev version; assume the version is of the form
+    # major[.minor[.patchlevel]], and increment the patchlevel by 1.
+    (major, minor, patchlevel) = (tag + ".0.0").split(".")[
+        :3
+    ]  # because minor and/or patchlevel are optional
```

Comment:
Do you think it'll be worth moving the comment so it doesn't split the slicing like this?

Reply:
Agreed!